# NBA Multiple Linear Regression - Statsmodels

In [1]:
# import required modules
import pandas as pd
from sklearn import linear_model
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.metrics import classification_report, confusion_matrix
import pandas as pd
from sklearn.metrics import roc_auc_score
from sklearn.metrics import precision_recall_fscore_support
import numpy as np
import statsmodels.api as sm

In [2]:
# load dataset
data = pd.read_csv('../../Capstone/Data/NBApipeline1.csv', header=0)
data.shape

(24851, 27)

In [3]:
data.isna().mean().round(4)*100

Player    0.0
Pos       0.0
Tm        0.0
Season    0.0
G         0.0
MP        0.0
PER       0.0
USG%      0.0
Age       0.0
3PAr      0.0
FTr       0.0
OWS       0.0
DWS       0.0
WS        0.0
WS/48     0.0
OBPM      0.0
DBPM      0.0
BPM       0.0
VORP      0.0
TS%       0.0
ORB%      0.0
DRB%      0.0
TRB%      0.0
AST%      0.0
STL%      0.0
BLK%      0.0
TOV%      0.0
dtype: float64

In [4]:
data.columns

Index(['Player', 'Pos', 'Tm', 'Season', 'G', 'MP', 'PER', 'USG%', 'Age',
       '3PAr', 'FTr', 'OWS', 'DWS', 'WS', 'WS/48', 'OBPM', 'DBPM', 'BPM',
       'VORP', 'TS%', 'ORB%', 'DRB%', 'TRB%', 'AST%', 'STL%', 'BLK%', 'TOV%'],
      dtype='object')

In [5]:
features=['USG%', 'Age','FTr', 'DBPM', 'ORB%', 'DRB%', 'TRB%', 'AST%', 'BLK%', 'TOV%']
X=data[features]
y=data.WS

In [6]:
# split X and y into training and testing sets
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.25,random_state=2019)

In [7]:
model = sm.OLS(y, X).fit()
predictions = model.predict(X)
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                     WS   R-squared (uncentered):                   0.273
Model:                            OLS   Adj. R-squared (uncentered):              0.272
Method:                 Least Squares   F-statistic:                              931.3
Date:                Mon, 22 Jun 2020   Prob (F-statistic):                        0.00
Time:                        01:48:43   Log-Likelihood:                         -31051.
No. Observations:               24851   AIC:                                  6.212e+04
Df Residuals:                   24841   BIC:                                  6.220e+04
Df Model:                          10                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
USG%           0.1928      0.007     26.552      0.000       0.179       0.207
Age            0.1227      0.005     22.586      0.000       0.112       0.133
FTr            0.2019      0.006     35.813      0.000       0.191       0.213
DBPM           0.2156      0.006     35.075      0.000       0.204       0.228
ORB%           0.0913      0.010      9.141      0.000       0.072       0.111
DRB%           0.0875      0.012      7.198      0.000       0.064       0.111
TRB%          -0.0539      0.013     -4.058      0.000      -0.080      -0.028
AST%           0.2710      0.006     43.733      0.000       0.259       0.283
BLK%           0.1040      0.008     12.977      0.000       0.088       0.120
TOV%          -0.4123      0.007    -58.845      0.000      -0.426      -0.399
==============================================================================
Omnibus:                      257.940   Durbin-Watson:                   1.780
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              240.630
Skew:                          -0.205   Prob(JB):                     5.59e-53
Kurtosis:                       2.747   Cond. No.                         6.12
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Equation:
WS = 0.1928(USG%) + 0.1227(Age) + 0.2019(FTr) + 0.2156(DBPM) + 0.0913(ORB%) + 0.0875(DRB%) - 0.0539(TRB%) + 0.2710(AST%) + 0.1040(BLK%) - 0.4123(TOV%)